pkg/portal/cluster/statistics.go

```diff
@@ -0,0 +1,115 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"time"
+
+	prometheusAPI "github.com/prometheus/client_golang/api"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+)
+
+type MetricValue struct {
+	Timestamp time.Time `json:"timestamp"`
+	Value     float64   `json:"value"`
+}
+
+type Metrics struct {
+	Name  string        `json:"metricname"`
+	Value []MetricValue `json:"metricvalue"`
+}
+
+func (c *client) Statistics(ctx context.Context, httpClient *http.Client, promQuery string, duration string, endTime time.Time) ([]Metrics, error) {
+	return c.fetcher.Statistics(ctx, httpClient, promQuery, duration, endTime)
+}
+
+func (f *realFetcher) Statistics(ctx context.Context, httpClient *http.Client, promQuery string, duration string, endTime time.Time) ([]Metrics, error) {
+	promConfig := prometheusAPI.Config{
+		Address: "http://prometheus-k8s-0:9090",
```

Comment:
couldn't we reuse the domain name from the director function here?

Reply:
where is this domain in the director function? :thinking:

Reply:
```go
	r.RequestURI = ""
	r.URL.Scheme = "http"
	r.URL.Host = "prometheus-k8s-0:9090"
	r.URL.Path = "/" + strings.Join(strings.Split(r.URL.Path, "/")[10:], "/")
	r.Header.Del("Cookie")
	r.Header.Del("Referer")
	r.Host = r.URL.Host
```

Reply:
line 58-64

Reply:
Okay, I get it why you want me to have it from there, to keep this URL at the single place and we don't have to explicitly change it everywhere when the URL will get changed. 

However, I will have to change the director function itself, in order to not to panic as the resourceID grepping logic is making the program panic, and then will have to extract the URL from the request.
So, why can't we create a new Function which actually gives this domain(to the Director functions as well)??

Reply:
that's a good idea :+1:

Reply:
Done.